### PR TITLE
Move messageCount exception to getMessageCount() method

### PR DESCRIPTION
**Pull request by chrisminett**
Allows the constructor to complete, and the other properties be read

### CODE REVIEW

**Comment by Pudge601** [comment]
I find the premise of this PR quite weird.. is there a particular use case this is allowing for?

It feels odd to me to throw the exception when trying to get the message count.

If we consider it to be invalid for the message count to be greater than the maximum message count, then I would generally expect the entire value object to be invalid, and therefore it makes sense that the exception is being thrown in the constructor to prevent the creation of an immutable object with an invalid state.

I guess it could be argued that we don't consider an invalid message count to imply an invalid state for the value object as a whole, but that brings me back to why I wonder if there is actually a use case for having a partially invalid value object.

**Comment by chrisminett** [comment]
The current problem is that throwing the exception in the constructor stops an object being returned, which prevents using `getSize()` to find the current number of characters (and therefore display to the user how much they're over the allowed max).

Perhaps an alternative would be to introduce a `validate()` method which will throw this exception, so that even `getMessageCount()` would be able to return a value >255

**Comment by Pudge601** [comment]
Yeah, I think I prefer that option.

**Comment by chrisminett** [comment]
bump

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 should be provided as a UTF-8 string. In most cases this should not require any
 implementation changes as UTF-8 is the
 [default charset for PHP](http://php.net/manual/en/ini.core.php#ini.default-charset).
+- The exception thrown for message content which exceeds the maximum allowed
+number of concatenated SMS (255) has been moved from the inspection on construct
+to the `getMessageCount()` method. This allows the implementation to still read
```
Comment (Pudge601) [inline CHANGELOG.md]: This changelog entry needs updating based on the new behaviour

**Review by Pudge601 — APPROVED** [review]

**Comment by chrisminett** [comment]
@Pudge601 do we class this as a BC break, requiring a new major version, or a feature, requiring a new minor version?

I don't much like a jump already to *v2.0.0*, especially when we're the only ones using this. The API stays the same, however for any implementation specifically depending upon the message length exception, it will have to be changed to call the new `validate()` method.

**Comment by Pudge601** [comment]
I think you'd have to be speaking very strictly to consider this change in behaviour a change to what we consider the public API of this library..

So yeah, I say just do it as a minor version, and hope the SemVer gods are looking the other way ;)

**Pull request merged**